snowcap/api/rust/src/widget.rs

```diff
@@ -51,6 +53,35 @@ impl From<u32> for WidgetId {
     }
 }
 
+/// A unique key for a widget.
+///
+/// While most widget only need a [`WidgetId`], some need a stable identifier across view
+/// generations if multiple events can fire at once. WidgetKey holds both identifier in a way that
+/// can be used as a key in a HashMap.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub(crate) enum WidgetKey {
+    WidgetId(WidgetId),
+    UniqueId(String),
+}
+
+impl Default for WidgetKey {
+    fn default() -> Self {
+        Self::WidgetId(Default::default())
+    }
+}
```

Comment:
re merging `unique_id` and `widget_id` as a new protobuf `message`:
At least on the rust side, this would mean the internal repr and the protobuf one would be coherent.